### PR TITLE
Week 2 Ready for Review (Project Restructure & Add Exercises)

### PR DESCRIPTION
**Pull request by mybffjustin**
Week 2 progress ready for review @matty377

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -188,6 +188,26 @@ <h1 class = "arrow">Education</h1>
             <div class = "row">
                 <div class = "col-md-12">
                     <div class = "education">
+                        <!-- Education Box 0 -->
+                        <div class = "col-md-4 wp2 item">
+                            <div class = "icon">
+                                <i class = "fa fa-pencil"></i><!-- Icon -->
```
Comment (matty377) [inline portfolio/src/main/webapp/index.html:192]: `<i>` tags aren't really used much anymore. They're for stand-out or idiomatic text and the alternatives listed here are typically the inline elements used instead - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/i

File: portfolio/src/main/java/com/google/sps/controller/RequestCounterServlet.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        // save counter to the application scope
+        getServletContext().setAttribute("counter", count);
+    }
+
+    protected void doGet(HttpServletRequest request,
+                         HttpServletResponse response) throws
+                                                                                   ServletException,
+                                                                                   IOException {
+        ServletContext servletContext = getServletContext();
+        // get the counter
```
Comment (matty377) [inline portfolio/src/main/java/com/google/sps/controller/RequestCounterServlet.java:29]: Comments like this are typically unnecessary since the line of code is actually quite readable! I can already tell what the code does without the comment.

Comments describing behavior like this are also error prone because someone might come along and change the code and not the comment or vice versa.

Same goes for most of the comments after this one, like `// increment the counter`

TL;DR - Readable code is usually self-documenting.

File: portfolio/src/main/java/com/google/sps/controller/HttpRequestServlet.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+@WebServlet(name = "httpRequestServlet",
+            urlPatterns = {"/httpRequestServlet"})
+public class HttpRequestServlet extends HttpServlet {
+
+    /**
+     * Handles HTTP GET requests.
```
Comment (matty377) [inline portfolio/src/main/java/com/google/sps/controller/HttpRequestServlet.java:16]: Nice Javadoc

File: portfolio/src/main/java/com/google/sps/controller/HttpRequestServlet.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                      HttpServletResponse response) throws
+                                                                                ServletException,
+                                                                                IOException {
+        response.setContentType("text/html");
+
+        PrintWriter out = response.getWriter();
+        System.out.println("Is this logging? (HttpRequestServlet.java) (bad " +
+                           "log version)");
+        log("Is this logging? (HttpRequestServlet.java)");
+        out.print("<HTML>");
```
Comment (matty377) [inline portfolio/src/main/java/com/google/sps/controller/HttpRequestServlet.java:38]: nit - Any reason these are uppercase?

**Review by matty377 — APPROVED** [review]
I did not really review the restructuring commits but the rest LGTM.

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (mybffjustin) [reply portfolio/src/main/webapp/index.html:192]: Thank you for letting me know about this🙏🙏🙏

File: portfolio/src/main/java/com/google/sps/controller/RequestCounterServlet.java (hunk 2 above)
Comment (mybffjustin) [reply portfolio/src/main/java/com/google/sps/controller/RequestCounterServlet.java:29]: Oh, I see, that makes a lot of sense! Thank you for helping me improve🙏🙏🙏

File: portfolio/src/main/java/com/google/sps/controller/HttpRequestServlet.java (hunk 4 above)
Comment (mybffjustin) [reply portfolio/src/main/java/com/google/sps/controller/HttpRequestServlet.java:38]: I am not sure why I made these uppercase. I will keep tags in lowercase moving forward.
